packages/authjs-nuxt/src/runtime/lib/server.ts

```diff
@@ -21,12 +21,16 @@ if (!globalThis.crypto) {
 /**
  * This is the event handler for the catch-all route.
  * Everything can be customized by adding a custom route that takes priority over the handler.
- * @param options AuthConfig
+ * @param options AuthConfig|Function
  * @param runtimeConfig RuntimeConfig
  * @returns EventHandler
  */
-export function NuxtAuthHandler(options: AuthConfig, runtimeConfig: RuntimeConfig) {
+export function NuxtAuthHandler(options: AuthConfig|Function, runtimeConfig: RuntimeConfig) {
```

Comment:
We need some proper typescript support for this union type.

Reply:
Thanks, just pushed up an update with some fixes for the linting errors too. I'm fairly new to Typescript so let me know if I need to make any more changes. Naming things is also always a pain, so feel free to correct stuff if you prefer.